docs/erofs-layers.md

```diff
@@ -0,0 +1,140 @@
+*   **End-to-End Integrity:** Using DM-Verity, the integrity of the filesystem can be verified at the block level by the kernel during runtime, providing stronger security guarantees than file-level checksums.
+*   **Efficient Distribution:** Distributing the filesystem image directly avoids local conversion steps (e.g., unpacking tarballs and creating filesystems), reducing startup latency and CPU usage.
+
+## 2. Layer Formats
+
+This specification defines two primary layer formats: **Uncompressed** and **Compressed (Zstd)**. Both formats support optional DM-Verity data for runtime integrity.
+
+### 2.1 Uncompressed EROFS Layer
+
+This layer consists of a standard EROFS filesystem image. It may optionally append DM-Verity hash tree data at the end of the blob.
+
+**Structure:**
+
+```text
++-----------------------------------------+
+|                                         |
+|  EROFS Filesystem Image                 |
+|  (Standard EROFS superblock & data)     |
+|                                         |
++-----------------------------------------+
+|                                         |
+|  DM-Verity Hash Data (Optional)         |
+|  (Appended at the end)                  |
+|                                         |
++-----------------------------------------+
+```
+
+### 2.2 Compressed EROFS Layer (Zstd)
+
+This layer uses Zstd compression to reduce transfer size. Unlike standard compressed layers (like `.tar.gzip`), this format supports random access by utilizing a **Chunk Mapping Table**.
+
+The blob consists of:
+1.  **Zstd Compressed Frames:** The raw EROFS filesystem image data compressed in standard [Zstd frames](https://github.com/facebook/zstd/blob/dev/doc/zstd_compression_format.md#frames).
+2.  **Chunk Mapping Table:** A custom table stored inside a [Zstd Skippable Frame](https://github.com/facebook/zstd/blob/dev/doc/zstd_compression_format.md#skippable-frames). This allows standard Zstd decompressors to ignore it, while aware readers can use it to locate specific uncompressed chunks.
+3.  **DM-Verity Hash Data (Optional):** Appended at the end.
+
+**Structure:**
+
+```text
++-----------------------------------------+ <--- Start of Blob
+|                                         |
+|  Zstd Compressed Frames                 |
+|  (Compressed Raw EROFS Image)           |
+|                                         |
++-----------------------------------------+ <--- Offset via Annotation
+|                                         |
+|  Skippable Zstd Frame (0x184D2A5E)      |
+|  +-----------------------------------+  |
+|  | Chunk Mapping Table               |  |
+|  +-----------------------------------+  |
+|                                         |
++-----------------------------------------+ <--- Offset via Annotation
+|                                         |
```

Comment:
can we move this part prior to `Chunk Mapping Table`?
I think `raw erofs image` and `DM-Verity Hash Data` should be as the whole component, and chunk mapping table can be regarded as OOB data, for lazy pulling for example